scanpy/plotting/_anndata.py

```diff
@@ -1939,103 +1887,170 @@ def dotplot(
         mean_obs = mean_obs.iloc[dendro_data['categories_idx_ordered'], :]
         fraction_obs = fraction_obs.iloc[dendro_data['categories_idx_ordered'], :]
 
-        y_ticks = range(mean_obs.shape[0])
-        dendro_ax = fig.add_subplot(axs[1, 1], sharey=dot_ax)
-        _plot_dendrogram(
-            dendro_ax, adata, groupby, dendrogram_key=dendrogram, ticks=y_ticks
+        dendro_ticks = np.arange(mean_obs.shape[0]) + 0.5
+
+    category_height = category_width = 0.35
+    legends_width = 1.2
+    dendro_width = 0.8 if dendrogram else 0
+    if figsize is None:
```

Comment:
Dotplot (and matrixplot, heatmap) by default override the ` rcParams['figure.figsize']`, otherwise the output figure will look very bad. I know that this is a different behavior compared to `sc.pl.scatter`.

The solution is to pass `figsize` as argument in case you want to use custom values. 

Do you have any other idea?